src/vs/base/browser/ui/highlightedlabel/highlightedLabel.ts

```diff
@@ -44,14 +44,51 @@ export class HighlightedLabel implements IDisposable {
 			highlights = [];
 		}
 
+		const originalHighlights = [...highlights];
 		this.text = text;
 		this.highlights = highlights;
+		if (escapedNewLines) {
+			this.highlights = this.adjustHighlightsForEscapedLineBreaks(this.highlights);
+		}
 		this.render();
+		this.highlights = originalHighlights;
+	}
+
+	private adjustHighlightsForEscapedLineBreaks(highlights: IHighlight[]): IHighlight[] {
```

Comment:
Maybe I am missing something but this method finds newlines and adjusts highlight ranges but where is the text being modified? We want the label to render a newline-symbol instead of reading on multiple lines, right?